core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -2740,10 +2744,14 @@ class WorkspaceService(protected val ctx: RawlsRequestContext,
           petKey,
           expectedGoogleBucketPermissions
         )
-        .recoverWith { case t: StorageException =>
+        .recoverWith {
           // Throw with the status code of the exception (for example 403 for invalid billing, 400 for requester pays)
           // instead of a 500 to avoid Sentry notifications.
-          Future.failed(new RawlsExceptionWithErrorReport(ErrorReport(t.getCode, t)))
+          case t: StorageException =>
+            Future.failed(new RawlsExceptionWithErrorReport(ErrorReport(getStatusCodeHandlingUnknown(t.getCode), t)))
+          case t: GoogleJsonResponseException =>
```

Comment:
It's looking like I should always expect the `StorageException` to be wrapped in a `GoogleJsonResponseException`, but I figure handling both isn't a big deal.

The 2 different stack traces showing here are pretty confusing to me-- it wasn't clear to me that StorageException wasn't the thing to handle: https://broad-institute.sentry.io/issues/3915484154/?project=153297&query=is%3Aignored&referrer=issue-stream&stream_index=1

Reply:
That is really confusing, agreed it doesn't hurt to catch both.